File: packages/xpute/core/src/math/vec3.ts
// @xpute/core/math/vec3.ts

import type { f64 } from "../abi/word.ts";

/** A point or a direction as three doubles, written into by the matrix
 * functions and read by whatever asked. */
export interface Vec3 {
  x: f64;
  y: f64;
  z: f64;
}

File: crates/xpute/core/src/math/vec3.rs
// xpute-core/math/vec3.rs

//! A point or a direction as three doubles, written into by the matrix
//! functions and read by whatever asked.

#[derive(Clone, Copy, Debug, Default, PartialEq)]
pub struct Vec3 {
    pub x: f64,
    pub y: f64,
    pub z: f64,
}
